src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/PublishContent.targets

```diff
@@ -10,13 +10,18 @@
   <!-- gathers the items to be published -->
   <Target Name="GatherItemsForPattern">
     <Error Condition="'$(PublishPattern)' == ''" Text="Please specify a value for PublishPattern using standard msbuild 'include' syntax." />
+
+    <PropertyGroup>
+      <ReleativeBlobPathBase>$(BlobNamePrefix)</ReleativeBlobPathBase>
+      <ReleativeBlobPathBase Condition="'$(ReleativeBlobPathBase)' != '' and !HasTrailingSlash('$(ReleativeBlobPathBase)')">$(ReleativeBlobPathBase)</ReleativeBlobPathBase>
+    </PropertyGroup>
     <ItemGroup>
       <ForPublishing Include="$(PublishPattern)" />
     </ItemGroup>
     <!-- add relative blob path metadata -->
     <ItemGroup>
       <ForPublishing>
-        <RelativeBlobPath>$([System.String]::Copy('%(RecursiveDir)%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath> 
+        <RelativeBlobPath>$(ReleativeBlobPathBase)$([System.String]::Copy('%(RecursiveDir)%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath>
```

Comment:
I can already envision this having challenges when BlobNamePrefix doesn't end with '/', do we want to perhaps append one in the case where it doesn't come with that premptively?